core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/chunkedencoding/AwsChunkSigner.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.chunkedencoding;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * Represents a signer for a chunk of data, that returns a new signature based on the data and the
+ * previous signature. This signature should be a string of hex characters.
+ */
+@SdkProtectedApi
+public interface AwsChunkSigner {
+
+    String signChunk(byte[] chunkData, String previousSignature);
+
+    String signChecksumChunk(byte[] calculatedChecksum, String previousSignature, String checksumHeaderForTrailer);
+}
```

Comment:
I don't mind this as a protected API, but would it be less brittle if we didn't try to abstract chunked signing? We could instead build a protected API for general processing/transformation of chunk-encoded streams (that would be useful for compression and checksums as well) and a particular signer could use that for chunked signing?